app/importer/server/classes/ImportDataConverter.ts

```diff
@@ -0,0 +1,832 @@
+			fields: {
+				_id: 1,
+				username: 1,
+			},
+		};
+
+		if (importId === 'rocket.cat') {
+			return {
+				_id: 'rocket.cat',
+				username: 'rocket.cat',
+			};
+		}
+
+		if (this._userCache.has(importId)) {
+			return this._userCache.get(importId) as IUserIdentification;
+		}
+
+		const user = Users.findOneByImportId(importId, options);
+		if (user) {
+			return this.addUserToCache(importId, user._id, user.username);
+		}
+
+		return null;
+	}
+
+	findImportedUserId(_id: string): string | undefined {
+		const data = this.findImportedUser(_id);
+		return data?._id;
+	}
+
+	findImportedUsername(_id: string): string | undefined {
+		const data = this.findImportedUser(_id);
+		return data?.username;
+	}
+
+	findImportedUserDisplayName(importId: string): string | undefined {
+		const options = {
+			fields: {
+				_id: 1,
+				name: 1,
+				username: 1,
+			},
+		};
+
+		if (this._userDisplayNameCache.has(importId)) {
+			return this._userDisplayNameCache.get(importId);
+		}
+
+		const user = importId === 'rocket.cat' ? Users.findOneById('rocket.cat', options) : Users.findOneByImportId(importId, options);
+		if (user) {
+			if (!this._userCache.has(importId)) {
+				this.addUserToCache(importId, user._id, user.username);
+			}
+
+			return this.addUserDisplayNameToCache(importId, user.name);
+		}
+	}
+
+	updateRoomId(_id: string, roomData: IImportChannel): void {
+		const set = {
+			ts: roomData.ts,
+			topic: roomData.topic,
+			description: roomData.description,
+		};
+
+		const roomUpdate: {$set?: Record<string, any>; $addToSet?: Record<string, any>} = {};
+
+		if (Object.keys(set).length > 0) {
+			roomUpdate.$set = set;
+		}
+
+		if (roomData.importIds.length) {
+			roomUpdate.$addToSet = {
+				importIds: {
+					$each: roomData.importIds,
+				},
+			};
+		}
+
+		if (roomUpdate.$set || roomUpdate.$addToSet) {
+			Rooms.update({ _id: roomData._id }, roomUpdate);
+		}
+	}
+
+	getRoomCreatorId(roomData: IImportChannel, startedByUserId: string): string {
+		if (roomData.u) {
+			const creatorId = this.findImportedUserId(roomData.u._id);
+			if (creatorId) {
+				return creatorId;
+			}
+
+			if (roomData.t !== 'd') {
+				return startedByUserId;
+			}
+
+			throw new Error('importer-channel-invalid-creator');
+		}
+
+		if (roomData.t === 'd') {
+			for (const member of roomData.users) {
+				const userId = this.findImportedUserId(member);
+				if (userId) {
+					return userId;
+				}
+			}
+		}
+
+		throw new Error('importer-channel-invalid-creator');
+	}
+
+	insertRoom(roomData: IImportChannel, startedByUserId: string): void {
+		// Find the rocketchatId of the user who created this channel
+		const creatorId = this.getRoomCreatorId(roomData, startedByUserId);
+		const members = this.convertImportedIdsToUsernames(roomData.users, roomData.t !== 'd' ? creatorId : undefined);
+
+		if (roomData.t === 'd') {
+			if (members.length < roomData.users.length) {
+				console.warn('One or more imported users not found: ${ roomData.users }');
```

Comment:
thx to LTGM.. this log will not show what is expected =)

btw, any reason to not use Logger?

Reply:
Not anymore, just didn't update this part of the code.